src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/GCHandleTests.cs

```diff
@@ -146,7 +146,7 @@ public void Equals_Object_ReturnsExpected(GCHandle handle, object other, bool ex
             finally
             {
                 handle.Free();
-                if (other is GCHandle otherHandle)
+                if (other is GCHandle otherHandle && !expected)
```

Comment:
> ```!ReferenceEquals(handle, otherHandle)```

`GCHandle` is a struct... I don't think it will work as expected with `ReferenceEquals` as it will be comparing to a boxed instance.